### PR TITLE
SystemTimer fixups

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description

Resolved TODOs, add `wait_until` to target alarm, add `&mut self` to public API methods.

Add https://github.com/esp-rs/esp-pacs/pull/229 to fixup the esp32s2 registers

### CODE REVIEW

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/systimer.rs [hunk 1]
```diff
@@ -1,29 +1,27 @@
 //! # System Timer peripheral driver
 //!
-//! ## Overview
-//! This software module provides an interface to interact with the system timer
-//! (SYSTIMER) peripheral on ESP microcontroller chips.
+//! The System Timer is a
+#![cfg_attr(esp32s2, doc = "`64-bit`")]
```
Comment (bjoernQ) [inline esp-hal/src/systimer.rs:4]: ❤️

**Review by bjoernQ — APPROVED** [review]
Thanks! Looks good to me

**Review by SergioGasquez — APPROVED** [review]
LGTM, thanks!

**Review by jessebraham — APPROVED** [review]

**Pull request merged**